### PR TITLE
[19263] User configuration for SHM metatraffic

### PR DESCRIPTION
**Pull request by MiguelCompany**
Adds documentation for eProsima/Fast-DDS#3753

@mergifyio backport 2.11.x 2.10.x 2.6.x

### CODE REVIEW

File: docs/fastdds/property_policies/non_consolidated_qos.rst [hunk 1]
```diff
@@ -303,8 +303,10 @@ Currently there are two different exchange formats which can be selected using t
 SHM transport meta-traffic enforcement
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-A DomainParticipant will by default configure both a :ref:`transport_udp_udp` and a :ref:`transport_sharedMemory_sharedMemory`.
-When a participant on another process in the same host is discovered, the endpoint discovery might be done using either transport.
+A DomainParticipant will by default configure both a :ref:`transport_udp_udp` and a
+:ref:`transport_sharedMemory_sharedMemory`. When a participant on another process in the same host is
```
Comment (EduPonz) [inline docs/fastdds/property_policies/non_consolidated_qos.rst:307]:
```suggestion
:ref:`transport_sharedMemory_sharedMemory`.
When a participant on another process in the same host is
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: code/DDSCodeTester.cpp [hunk 2]
```diff
@@ -787,6 +787,17 @@ void dds_domain_examples()
             "true");
         //!--
     }
+
+    {
+        // DDS-SHM-ENFORCE-META-TRAFFIC
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:792]:
```suggestion
        //DDS-SHM-ENFORCE-META-TRAFFIC
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]
LGTM with green CI

**Comment by MiguelCompany** [comment]
@Mergifyio backport 2.11.x 2.10.x 2.6.x

**Pull request merged**